### ISSUE TITLE
How to force fresh start on specific app version?

### ISSUE THREAD

**Issue by outaTiME** [rt2zz/redux-persist#578]
Hi guys, im building react native app with redux-persist v5 and I would like to know how to force fresh start on specific app version ignoring any previous store data.
Im trying to make something with `createMigrate` but it execute every time and make transformations over stored data.

For example if my app version is 2.0 I want to execute purge only once forcing to user to put their credentials again ...

any feedback will be appreciated, thanks !!!

**Comment by outaTiME** [comment]
(for now) To solve this what I did  was to change the configuration key to force fresh start to 0.2.1 version, but it would be ideal to execute purge (one time) to remove all the data saved previously, here the code:

<img width="584" alt="screen shot 2017-11-20 at 01 05 58" src="https://user-images.githubusercontent.com/140472/33001906-ab327f48-cd8f-11e7-9d4d-e7d512c399af.png">

**Comment by rt2zz** [comment]
this is what `createMigrate` is intended for. It should only run migrations against older versions of state. You were observing the same migration running every time?

**Comment by outaTiME** [comment]
Hi @rt2zz, I need to clean my storage for a specific version and only once.
I’m using the following code:

```js
const migrations = {
  1: (state) => {
    return {};
  }
}
const config = {
  key: 'root',
  storage: storage,
  version: 1,
  whitelist: ['user', 'roadmaps'],
  transforms: [
    encryptor
  ],
  debug: __DEV__,
  migrate: createMigrate(migrations, { debug: __DEV__ }),
}
```

The problem is this is executed every time and I need to clean my storage only once.

**Comment by dziamid** [comment]
@outaTiME 
```
const migrations = {
  1: ({_persist}) => {
    return {_persist}; // *reset all state, except version*
  }
}
const config = {
  key: 'root',
  storage: storage,
  version: 1,
  whitelist: ['user', 'roadmaps', '_persist'], // *persist state version*
  transforms: [
    encryptor
  ],
  debug: __DEV__,
  migrate: createMigrate(migrations, { debug: __DEV__ }),
}
```

**Comment by outaTiME** [comment]
Great @dziamid, but @rt2zz what about this? this should not be handled automatically by the migrator?

**Comment by ScreamZ** [comment]
persistCombineReducers should, yes....

**Comment by outaTiME** [comment]
@rt2zz any way to deal with that? it could be possible? i need to achive that using @dziamid way?

thanks !!!

**Comment by rt2zz** [comment]
@outaTiME @dziamid yes this should be handled by the library. I think this was inadvertently broken when simplifying some of the persistReducer code. I will have to take a look later this week - hopefully something we can add a test for in the future.
